packages/preview-middleware-client/src/adp/quick-actions/fe-v4/table-quick-action-base.ts

```diff
@@ -19,6 +20,7 @@ export abstract class TableQuickActionDefinitionBase extends QuickActionDefiniti
     isClearButtonEnabled = false;
     children: NestedQuickActionChild[] = [];
     tableMap: Record<string, number> = {};
+    smartTables: UI5Element[] = [];
```

Comment:
Instead of making a new array, it would be better to use `tableMap` like in https://github.com/SAP/open-ux-tools/blob/408a1c5e36714a5fb99ce6602cf1c845d3ee8222/packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts#L61-L74 or even better use that base class instead.

Reply:
Refactored